crates/sui/src/keytool.rs

```diff
@@ -683,51 +868,242 @@ impl KeyToolCommand {
+                })
             }
-        }
-        Ok(())
+        });
+
+        cmd_result
     }
 }
 
-fn convert_string_to_base64(value: String) -> Result<String, anyhow::Error> {
+fn convert_private_key_to_base64(value: String) -> Result<String, anyhow::Error> {
     match Base64::decode(&value) {
         Ok(decoded) => {
-            assert_eq!(decoded.len(), 33);
-            let hex_encoded = Hex::encode(&decoded[1..]);
-            info!("Wallet formatted private key: 0x{}", hex_encoded);
-            Ok(hex_encoded)
+            if decoded.len() != 33 {
+                return Err(anyhow!(format!("Private key is malformed and cannot base64 decode it. Fed 33 length but got {}", decoded.len())));
+            }
+            Ok(Hex::encode(&decoded[1..]))
         }
         Err(_) => match Hex::decode(&value) {
             Ok(decoded) => {
-                assert_eq!(decoded.len(), 32);
+                if decoded.len() != 32 {
+                    return Err(anyhow!(format!("Private key is malformed and cannot hex decode it. Expected 32 length but got {}", decoded.len())));
+                }
                 let mut res = Vec::new();
                 res.extend_from_slice(&[SignatureScheme::ED25519.flag()]);
                 res.extend_from_slice(&decoded);
-                info!("Keystore formatted private key: {:?}", Base64::encode(&res));
                 Ok(Base64::encode(&res))
             }
-            Err(_) => {
-                info!("Invalid private key format");
-                Err(anyhow!("Invalid private key format"))
-            }
+            Err(_) => Err(anyhow!("Invalid private key format".to_string())),
         },
     }
 }
 
-fn store_and_print_keypair(address: SuiAddress, keypair: SuiKeyPair) {
-    let path_str = format!("{}.key", address).to_lowercase();
-    let path = Path::new(&path_str);
-    let address = format!("{}", address);
-    let kp = keypair.encode_base64();
-    let flag = keypair.public().flag();
-    let out_str = format!("address: {}\nkeypair: {}\nflag: {}", address, kp, flag);
-    fs::write(path, out_str).unwrap();
-    println!(
-        "Address, keypair and key scheme written to {}",
-        path.to_str().unwrap()
-    );
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Key {
+    sui_address: SuiAddress,
+    public_base64_key: String,
+    key_scheme: String,
+    flag: u8,
+    mnemonic: Option<String>,
+    peer_id: Option<String>,
+}
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PrivateKeyBase64 {
+    base64: String,
+}
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ZkLogInPrepare {
+    ephemeral_pubkey: String,
+    ephemeral_keypair: String,
+    nonce: String,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SignData {
+    sui_address: SuiAddress,
+    raw_tx_data: String,
+    intent: Intent,
+    raw_intent_msg: String,
+    digest: String,
+    sui_signature: String,
+}
```

Comment:
💯 to put this in a struct. i would add some comments to highlight what they mean:
```suggestion
pub struct SignData {
    sui_address: SuiAddress,
    raw_tx_data: String, // Base64 encoded string of serialized transaction data. 
    intent: Intent, // Intent struct used, see [struct Intent] for field definitions. 
    raw_intent_msg: String, // Base64 encoded [struct IntentMessage] consisting of (intent || message) where message can be `TransactionData` etc.
    digest: String, // Base64 encoded blake2b hash of the intent message, this is what the signature commits to. 
    sui_signature: String, // Base64 encoded `flag || signature || pubkey` for a complete serialized Sui signature to be send for executing the transaction. 
}
```